utils/docker/DOCKER_README.md

```diff
@@ -0,0 +1,124 @@
+Then we can run this image; a running Docker image is called a _container_.
+
+#### Basic Example
+
+```Docker
+FROM ubuntu:18.04
+COPY . /app
+RUN make /app
+RUN apt-get update && apt-get install -y vim git
+CMD python /app/app.py
+```
+
+Each instruction creates one layer:
+
+    * FROM creates a layer from the ubuntu:18.04 Docker image.
+    * COPY adds files from your Docker client’s current directory.
+    * RUN executes a system command in the Docker. Is compiled into the image, so always
+    * ENTRYPOINT/CMD:
+        execute a cmd in the started container.
+        * CMD: default arguments/cmd, can be overwritten when starting with `docker run $image $args`. If multiple CMD lines, only last one is considered.
+        * ENTRYPOINT: like CMD, but always executed.
+        So you should use ENTRYPOINT for calling executable and CMD for arguments that can be overwritten at container start.
+
+#### Other layers
+
+* COPY: copy from current directory into somewhere in the Docker image
+* WORKDIR: set active directory inside the image. Next like in Dockerfile will run from this WORKDIR, and when the container starts terminal opens here.
+
+---- 
+### Starting/running the container
+
+* Build/compile the Docker image: `docker build --tag <image_name> .`: compiles Dockerfile in current directory, and names it <image_name>. Don't forget the `.` at the end!
+    * the `docker_entrypoint.sh` file is executed when the container starts, so has access to the host file system via mounts (see launch.sh). It will set up ROS/catkin workspace.
+
+* Basic command: `docker run <image>`.
+    To go into interactive terminal inside the container, use `-it` argument (interactive).
+
+* start container in the background, return container id: `docker run -d` -> use example `docker_id=$(docker run -d -it | sed -n 2p)`
+* starting multiple terminals connected to the same container: `docker exec -it <container> bash`
+* attach terminal into running container: `docker attach ${docker_id}`
+
+* mount local directory inside the container (very useful for developing):
+    `docker run <image> -v <local_dir>:<container_dir>`
+
+    * by default Docker uses a root user. I create a user with UID/GID of your host user, so files created from inside Docker have correct permissions. For details check `Dockerfile` and `launch.sh`.
+
+* get access to host display inside container: `docker run -it --env="DISPLAY=$DISPLAY"` and some messy X stuff for permissions. See `launch.sh`
+
+* get access to host network: `--net=host`
+
+* get access to host GPU inside container: `docker run -it --gpus all --ipc=host`
+    * You might need to install [nvidia-docker2](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker)
+
+* you should put all these things in a bash script to launch the container, like `launch.sh`. 
+    * use some environment variable to make the launch script more flexible: `export WORKSPACE=<your_root_dir> && bash launch.sh `
```

Comment:
The tip to pass an environment variable is outdated - we discovered you can just pass an argument to the script.
So this sentence should be adapted.